proto/kava/auction/auction.proto

```diff
@@ -0,0 +1,77 @@
+syntax = "proto3";
+package kava.auction.v1beta1;
+
+import "gogoproto/gogo.proto";
+import "cosmos_proto/cosmos.proto";
+import "google/protobuf/timestamp.proto";
+
+option go_package = "github.com/kava-labs/kava/x/auction/types";
+
+option (gogoproto.goproto_getters_all) = false;
+
+// BaseAuction defines common attributes of all auctions
+message BaseAuction {
+  option (cosmos_proto.implements_interface) = "Auction";
+
+  uint64 id = 1;
+
+  string initiator = 2;
+
+  bytes lot = 3 [(gogoproto.customtype) = "github.com/cosmos/cosmos-sdk/types.Coin", (gogoproto.nullable) = false];
+
+  string bidder = 4;
+
+  bytes bid = 5 [(gogoproto.customtype) = "github.com/cosmos/cosmos-sdk/types.Coin", (gogoproto.nullable) = false];
```

Comment:
I'd use the cosmos.base.v1beta.Coin proto type instead of the gogo custom type.
The less custom marshalling we use the easier it is for clients to work with our proto files.

Reply:
https://github.com/Kava-Labs/kava/pull/1052/commits/e80008500f863b3feb07e87ce6926f0e27ba2bae